src/_pytest/reports.py

```diff
@@ -272,3 +407,21 @@ def __init__(self, msg):
 
     def toterminal(self, out):
         out.line(self.longrepr, red=True)
+
+
+def pytest_report_serialize(report):
+    if isinstance(report, (TestReport, CollectReport)):
+        data = report._to_json()
+        data["_report_type"] = report.__class__.__name__
+        return data
```

Comment:
Nitpick: these implementations don't actually serialize/unserialize the objects: they convert them to/from native Python types which can be serialised/deserialised as JSON. Serialisation, to my mind, means converting something to bytes (or at least str).

That's a fine API if that's what you want, but maybe the names should reflect it. Sorry for bikeshedding. ;-)

Reply:
Not at all, thanks for chipping in. I'm fine with changing those names now, easier to do this before merging. Do you have a suggestion?

Reply:
Maybe `pytest_report_to_json` and `pytest_report_from_json`? It's not strictly accurate either, because they're producing dicts suitable for JSON, not actual JSON, but I think that inaccuracy is widely accepted.

Otherwise:

- `pytest_report_to_dict` - accurate but less specific
- `pytest_report_to_jsonable` - kind of ugly, IMO

Reply:
I would like to avoid restricting this to "json", because we might change the actual representation to something else. 

I like `pytest_report_to_dict` and `pytest_report_from_dict`; I agree it is less accurate, but I believe it is enough to mention that it only supports built-in types in the docs.

Thanks a lot for the input! I will change to `pytest_report_to_dict` and `pytest_report_from_dict` later then. 👍

Reply:
at $previous_job we called this `def __primitive__(self):`

Reply:
JSON-compatible is a stricter standard than built-in types (e.g. no sets in JSON), but so long as the requirements are documented, I think the `dict` naming is a good option.

Reply:
What about `…_to_python` and `…_from_python`?  But it does not clearly indicate the overall direction (as with `…serialize` and `…unserialize`).

Reply:
`to/from_serializable` just to keep the door open for msgpack and yaml as well

Reply:
I like @RonnyPfannschmidt's, we describe the intent rather what the hook returns in detail. 👍

Reply:
Done, `pytest_report_from_serializable`/`pytest_report_to_serializable`. 👍 

If nobody objects I will merge this later then.